csi/controller_server.go

```diff
@@ -1069,19 +1070,28 @@ func (cs *ControllerServer) cleanupSnapshot(sourceVolumeName, id string) error {
 	return nil
 }
 
-func (cs *ControllerServer) cleanupBackupVolume(sourceVolumeName, id string) error {
+func (cs *ControllerServer) cleanupBackup(sourceVolumeName, id string) error {
 	backupVolumeName, backupName := sourceVolumeName, id
-	backupVolume, err := cs.getBackupVolume(backupVolumeName)
+	backupVolumes, err := cs.getBackupVolumes(backupVolumeName)
 	if err != nil {
 		return err
 	}
-	if backupVolume != nil && backupVolume.Name != "" {
-		if _, err = cs.apiClient.BackupVolume.ActionBackupDelete(backupVolume, &longhornclient.BackupInput{
-			Name: backupName,
-		}); err != nil {
-			return err
+	for _, bv := range backupVolumes {
+		// The BackupDelete API actually doesn't care about bv when deleting a backup.
+		// The bv is there for backward compatibility.
+		// Any bv will work
+		if bv.Name != "" {
+			_, err = cs.apiClient.BackupVolume.ActionBackupDelete(bv, &longhornclient.BackupInput{
+				Name: backupName,
+			})
+			if err != nil {
+				return err
+			} else {
+				return nil
+			}
```

Comment:
```diff
-    } else {
-       return nil
-    }
+    }
+
+    break
}
```

I think we don't need `else` here as the robot mentioned.
or we can return `err` directly, WDYT?

Reply:
Surely. Fixed. Thanks